### DISCUSSION TITLE
umi4中qiankun子应用如何在getInitialState中拿到主应用传值的数据？

### DISCUSSION THREAD

**Discussion by Chill7773** [umijs/umi#11827] [General]
主要是想要所有子应用共用主应用的initialState，想知道有没有比较好的解决方案

**Comment by fz6m** [comment]
把主应用自己想要的数据传给子应用，方式可参考 [父子应用通信](https://umijs.org/docs/max/micro-frontend#%E7%88%B6%E5%AD%90%E5%BA%94%E7%94%A8%E9%80%9A%E4%BF%A1) ，然后在想要的地方使用即可，如果只是为了一次性使用，可以使用其中第二种方式通过 qiankun 传递，当然使用全局 window 、状态管理库、总线 等方案都是可以的：

```ts
// app.ts

let props;

export const qiankun = {
  async mount(masterProps) {
    props = masterProps
  }
};

export const getInitialState = () => {
  // use props
}
```
